Update Compose file

king-alexander opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

Update the YAML configuration file to adhere to the [Docker Compose implementation](https://docs.docker.com/compose/intro/compose-application-model/) of the formal [Compose Specification](https://github.com/compose-spec/compose-spec).


## 💭 Motivation and context ##

<!-- Why is this change required? -->
The Compose Specification defines the currently supported version of the Docker Compose CLI (Compose V2). This change resolves #78. 
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

I tested this change locally with Docker version 4.40.0 (187762) and Docker Compose version v2.34.0-desktop.1. I verified that all services in the project started successfully, then ran existing unit tests. All passed.

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- These boxes should remain unchecked until the pull request has been -->
<!-- approved. -->

- [x] Finalize version.

## ✅ Post-merge checklist ##

<!-- Remove any of the following that do not apply. -->

- [x] Create a release.

king-alexander committed:
Update filename

Use the canonical name for a Compose file according to the
Compose Specification.

king-alexander committed:
Replace version property with name property

The `version` property is obsolete. The project name is
defined by the `name` property.

king-alexander committed:
Update README

Update the instructions for installation, execution, development,
and debugging to use the Compose V2 CLI syntax and correct the
reference to the Compose file.

king-alexander committed:
Add TODO comment

Reference #80 as future work.

king-alexander committed:
Bump version from 2.0.1 to 2.1.0

dav3r commented on compose.yaml:
```diff
@@ -1,7 +1,9 @@
+name: admiral
 
 secrets:
+  # TODO: https://github.com/cisagov/admiral/issues/80
+  # move out of secrets
```
Can this comment be clarified a bit?  Maybe something like:
```suggestion
  # TODO: Move config files from secrets to a new configs section
  # See https://github.com/cisagov/admiral/issues/80 for details
```

dav3r commented on README.md:
```diff
@@ -14,19 +14,19 @@ This project requires a [Docker](https://www.docker.com) installation.
   - `secrets/admiral.yml`
   - `secrets/redis.conf`
-  - `docker-compose.yml`
+  - `compose.yaml`
```
Today I learned that [`compose.yaml` is preferred to `compose.yml`](https://docs.docker.com/compose/intro/compose-application-model/#the-compose-file).  Since that is the case, we should change the other file extensions listed in this README as well (e.g. `admiral.yml` -> `admiral.yaml`, etc.).

king-alexander replied:
See [ecee5bf](https://github.com/cisagov/admiral/pull/79/commits/ecee5bf705f33ad0b1e7051d71b1b647f30819dc) and [6bed5ac](https://github.com/cisagov/admiral/pull/79/commits/6bed5ac774930eb16ec0ce51db5cbb607dc8d38f).

dav3r replied:
Looks good, thanks!

dav3r commented on README.md:
```diff
@@ -14,19 +14,19 @@ This project requires a [Docker](https://www.docker.com) installation.
   - [http://localhost:5555](http://localhost:5555)
 - Optional: Run the code tests
-  - `docker-compose -f docker-compose-dev.yml run test`
+  - `docker compose -f docker-compose-dev.yml run test`
```
`docker-compose-dev.yml` should be renamed `compose-dev.yaml` as part of this PR and the README should be updated accordingly.  Also, the same changes made to `compose.yaml` file should be applied to `compose-dev.yaml`.

king-alexander replied:
See [05e43a4](https://github.com/cisagov/admiral/pull/79/commits/05e43a488ca78fbbc9aec78d5d5f097ad0bfc09d), [ecee5bf](https://github.com/cisagov/admiral/pull/79/commits/ecee5bf705f33ad0b1e7051d71b1b647f30819dc), and [6bed5ac](https://github.com/cisagov/admiral/pull/79/commits/6bed5ac774930eb16ec0ce51db5cbb607dc8d38f).

dav3r reviewed (changes requested):
I have a few small items for your consideration.

king-alexander committed:
Clarify TODO comment

Co-authored-by: dav3r <david.redmin@gwe.cisa.dhs.gov>

king-alexander committed:
Update Compose development file

Rename file to the canonical form and replace the version
property with the name property.

king-alexander committed:
Update YAML extensions

Change `.yml` references to the preferred `.yaml` extension.

king-alexander committed:
Update README

Match references to renamed files and update YAML file extensions.

dav3r commented on README.md:
```diff
@@ -1,7 +1,7 @@
 
 [![GitHub Build Status](https://github.com/cisagov/admiral/workflows/build/badge.svg)](https://github.com/cisagov/admiral/actions)
-[![CodeQL](https://github.com/cisagov/admiral/workflows/CodeQL/badge.svg)](https://github.com/cisagov/admiral/actions/workflows/codeql-analysis.yml)
+[![CodeQL](https://github.com/cisagov/admiral/workflows/CodeQL/badge.svg)](https://github.com/cisagov/admiral/actions/workflows/codeql-analysis.yaml)
```
You got a little overzealous with your search and replace.  This file should remain called `codeql-analysis.yml` for now - if it ever changes, that will be done in our skeletons:
```suggestion
[![CodeQL](https://github.com/cisagov/admiral/workflows/CodeQL/badge.svg)](https://github.com/cisagov/admiral/actions/workflows/codeql-analysis.yml)
```

dav3r reviewed (approved):
Approved, with one small thang that should be reverted to it's original state.

king-alexander committed:
Revert CodeQL file extension

This file should retain the `.yml` extension unless changed upstream.

Co-authored-by: dav3r <david.redmin@gwe.cisa.dhs.gov>
